src/server/qgswfsserver.cpp

```diff
@@ -445,7 +445,7 @@ int QgsWFSServer::getFeature( QgsRequestHandler& request, const QString& format
     QDomElement queryElem;
     for ( int i = 0; i < queryNodes.size(); i++ )
     {
-      queryElem = queryNodes.at( 0 ).toElement();
+      queryElem = queryNodes.at( i ).toElement();
```

Comment:
Hi, I don't know exactly how it works, but I remember that mTypeNames preparation around line 448 were important and helps setting "summary" in xml/http response. Removing it cause missing "TYPENAME", e.g.

"""<wfs:FeatureCollection xmlns:wf...=1.0.0&amp;REQUEST=DescribeFeatureType&amp;TYPENAME=&amp;OUTPUTFORMAT=XMLSCHEMA"><gml:boundedBy>"""

instead of  

"""<wfs:FeatureCollection xmlns:wf...=1.0.0&amp;REQUEST=DescribeFeatureType&amp;TYPENAME=secondlayer,testlayer&amp;OUTPUTFORMAT=XMLSCHEMA"><gml:boundedBy>"""